### PR TITLE
Convert remaining ScopedDbServices to new Mongo API

### PR DESCRIPTION
**Pull request by kingzacko1**
## Description

- Refactor remaining `ScopedDbService` classes and their corresponding test classes.
- Remove now unused `ScopedDbService` class and corresponding test classes.

/nocl not user facing
/prd Graylog2/graylog-plugin-enterprise#7357
## Motivation and Context

Migrating DB services to use new Mongo API

## How Has This Been Tested?

local dev env. still working on testing everything but wanted to open this up so others could help since it touches quite a bit of code.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/events/rest/EventDefinitionsResource.java [hunk 1]
```diff
@@ -175,7 +176,7 @@ public PageListResponse<EventDefinitionDto> getPage(@ApiParam(name = "page") @Qu
                                                                   allowableValues = "title,description,priority,status")
                                                         @DefaultValue(DEFAULT_SORT_FIELD) @QueryParam("sort") String sort,
                                                         @ApiParam(name = "order", value = "The sort direction", allowableValues = "asc, desc")
-                                                        @DefaultValue(DEFAULT_SORT_DIRECTION) @QueryParam("order") String order) {
+                                                            @DefaultValue(DEFAULT_SORT_DIRECTION) @QueryParam("order") SortOrder order) {
```
Comment (kingzacko1) [inline graylog2-server/src/main/java/org/graylog/events/rest/EventDefinitionsResource.java:179]: @bernd what is the proper indentation level here? If I select Code > Reformat Code from the IntelliJ menu I get the line in red. But on save it reverts to the new line in green.

Comment (bernd) [reply graylog2-server/src/main/java/org/graylog/events/rest/EventDefinitionsResource.java:179]: @kingzacko1 Yeah, I would say the red line is the correct one. I'm not sure why it's changed on save.

Comment (kingzacko1) [reply graylog2-server/src/main/java/org/graylog/events/rest/EventDefinitionsResource.java:179]: Thanks! I'll figure it out and get consistent behavior in my IDE.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java [hunk 2]
```diff
@@ -258,25 +254,19 @@ public LookupTablePage tables(@ApiParam(name = "page") @QueryParam("page") @Defa
                                             allowableValues = "title,description,name,id")
                                   @DefaultValue(LookupTableDto.FIELD_TITLE) @QueryParam("sort") String sort,
                                   @ApiParam(name = "order", value = "The sort direction", allowableValues = "asc, desc")
-                                  @DefaultValue("desc") @QueryParam("order") String order,
+                                      @DefaultValue("desc") @QueryParam("order") SortOrder order,
```
Comment (danotorrey) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java:257]: Nit: Another strange indent

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java [hunk 3]
```diff
@@ -458,24 +448,18 @@ public DataAdapterPage adapters(@ApiParam(name = "page") @QueryParam("page") @De
                                               allowableValues = "title,description,name,id")
                                     @DefaultValue(DataAdapterDto.FIELD_TITLE) @QueryParam("sort") String sort,
                                     @ApiParam(name = "order", value = "The sort direction", allowableValues = "asc, desc")
-                                    @DefaultValue("desc") @QueryParam("order") String order,
+                                        @DefaultValue("desc") @QueryParam("order") SortOrder order,
```
Comment (danotorrey) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java:451]: Indent

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java [hunk 4]
```diff
@@ -673,24 +657,18 @@ public CachesPage caches(@ApiParam(name = "page") @QueryParam("page") @DefaultVa
                                        allowableValues = "title,description,name,id")
                              @DefaultValue(CacheDto.FIELD_TITLE) @QueryParam("sort") String sort,
                              @ApiParam(name = "order", value = "The sort direction", allowableValues = "asc, desc")
-                             @DefaultValue("desc") @QueryParam("order") String order,
+                                 @DefaultValue("desc") @QueryParam("order") SortOrder order,
```
Comment (danotorrey) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/lookup/LookupTableResource.java:660]: Indent

**Review by danotorrey — APPROVED** [review]
LGTM and spot-testing of event definitions, LUTs, and scope checks was successfully. Thanks a lot for tackling all of these conversions @kingzacko1.

**Review by ryan-carroll-graylog — APPROVED** [review]
LGTM and smoke tested the updated entities successfully, thanks for knocking this out!

**Pull request merged**